### ISSUE TITLE
Environment doesn't check global composer auth.json file when running `bin/magento sampledata:deploy`

### ISSUE THREAD

**Issue by schizek** [swiftotter/den#108]
**Version of Den**
1.0.0-beta.11

**Operating System and Installation Method**
Mac OS 12.6 via Brew

**Describe the bug**
While attempting to install a vanilla Magento2 install with sample data, the Den environment doesn't honor the composer creds already setup in my ~/.composer/auth.json file on my computer.  Instead it prompts for creds...

```
$ bin/magento sampledata:deploy
./composer.json has been updated
Running composer update magento/module-bundle-sample-data magento/module-widget-sample-data magento/module-theme-sample-data magento/module-wishlist-sample-data magento/module-sales-sample-data magento/module-tax-sample-data magento/module-sales-rule-sample-data magento/module-downloadable-sample-data magento/module-msrp-sample-data magento/module-catalog-sample-data magento/module-cms-sample-data magento/module-customer-sample-data magento/module-configurable-sample-data magento/module-product-links-sample-data magento/module-catalog-rule-sample-data magento/module-customer-balance-sample-data magento/module-grouped-product-sample-data magento/module-gift-registry-sample-data magento/module-gift-card-sample-data magento/module-target-rule-sample-data magento/module-review-sample-data magento/module-multiple-wishlist-sample-data magento/module-swatches-sample-data magento/module-offline-shipping-sample-data magento/sample-data-media
Loading composer repositories with package information
<warning>Warning from repo.magento.com: You haven't provided your Magento authentication keys. For instructions, visit https://devdocs.magento.com/guides/v2.3/install-gde/prereq/connect-auth.html</warning>
    Authentication required (repo.magento.com):
      Username:
```

**To Reproduce**
1. `den env-init testing magento2`
2. `den sign-certificate testing.test`
3. `den env up`
4. `den shell`
5. `composer create-project --repository-url=https://repo.magento.com/ magento/project-enterprise-edition /tmp/exampleproject 2.4.x`
6. `rsync -a /tmp/exampleproject/ /var/www/html/ && rm -rf /tmp/exampleproject/`
7. `bin/magento sampledata:deploy`

**Expected behavior**
It should run the `bin/magento sampledata:deploy` command without prompting for a login.

**Additional context**
Previously I've been running the DrPaynne fork and it would deploy sampledata without prompting

**Comment by navarr** [comment]
This was always my experience on Warden, though I never tested it on DrPayyne's. 

As I recall, Magento sets it's own composer home directory, but there's probably nothing stopping us from double mounting the composer config

**Comment by bap14** [comment]
@navarr Yes, Magento will use `<project_path>/var/composer_home` as its composer directory for certain tasks.

I would caution about defaulting to always mounting the current user's `~/.composer/auth.json` to the project auth.json. Doing so could cause problems for those devs that have a different auth.json for each project / client. I would prefer to see one or two flags that enable the mounting of `~/.composer/auth.json` to the project root and the Magento composer home.

**Comment by ihor-sviziev** [comment]
I think that behavior should be fixed on magento side, since it’s not the
default behavior of composer

**Comment by navarr** [comment]
Based on the responses from Brett and Ihor, and the fact that this is standard behavior, I'm going to close this issue as not planned.

I think they're right in that, Magento itself should use the base auth (why it sets its own composer home, I'll never know); and that mounting it is likely to cause issues.
